### PR TITLE
Set daemon to true when creating background thread pool, and also set…

### PR DESCRIPTION
**Pull request by Apache9**
… a proper name for the thread

Fix for #1057

### CODE REVIEW

File: jetcd-core/src/main/java/io/etcd/jetcd/impl/ClientConnectionManager.java [hunk 1]
```diff
@@ -73,7 +74,15 @@
         this.credential = new AuthCredential(this);
 
         if (builder.executorService() == null) {
-            this.executorService = Executors.newCachedThreadPool();
+            final ThreadFactory backingThreadFactory = Executors.defaultThreadFactory();
```
Comment (vinsia) [inline jetcd-core/src/main/java/io/etcd/jetcd/impl/ClientConnectionManager.java:77]: can you put this into https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/main/java/io/etcd/jetcd/support/Util.java to reuse?

Comment (Apache9) [reply jetcd-core/src/main/java/io/etcd/jetcd/impl/ClientConnectionManager.java:77]: Let me give it a try.

Comment (Apache9) [reply jetcd-core/src/main/java/io/etcd/jetcd/impl/ClientConnectionManager.java]: Done. PTAL?

**Review by vinsia — APPROVED** [review]
Thanks

**Pull request merged**